### PR TITLE
Modal widget adheres to dynamic height

### PR DESCRIPTION
**Pull request by riodeuno**
- This PR enables dynamic height for modal widget

### CODE REVIEW

File: app/client/src/sagas/dynamicHeightSagas.ts [hunk 1]
```diff
@@ -677,7 +715,7 @@ export default function* widgetOperationSagas() {
       batchCallsToUpdateWidgetDynamicHeightSaga,
     ),
     debounce(
-      200,
+      500,
```
Comment (riodeuno) [inline app/client/src/sagas/dynamicHeightSagas.ts:718]: Revert this change. 
I had added this to fix the undo-redo issue, but this doesn't help, and even if it did, this is not the right approach.

File: app/client/src/sagas/dynamicHeightSagas.ts [hunk 2]
```diff
@@ -500,22 +512,44 @@ export function* updateWidgetDynamicHeightSaga() {
         },
       ];
     }
+  }
 
-    log.debug("Dynamic height: Widgets to update:", { widgetsToUpdate });
+  if (Object.keys(independentUpdates).length > 0) {
+    Object.keys(independentUpdates).forEach((updatedWidgetId) => {
```
Comment (rahulramesha) [inline app/client/src/sagas/dynamicHeightSagas.ts:518]: Minor NIT: In between Object.keys and forEach, it has looping through independentUpdates 3 times. This i feel can be done in a single loop

**Review by rahulramesha — APPROVED** [review]

File: app/client/src/sagas/dynamicHeightSagas.ts (hunk 2 above)
Comment (riodeuno) [reply app/client/src/sagas/dynamicHeightSagas.ts:552]: Absolutely. Let me add a todo for this. I'll fix this today in a separate PR.
